root/io/pointers/restoreCanvas.C

```diff
@@ -0,0 +1,122 @@
+#include "TMemFile.h"
+#include "TCanvas.h"
+#include "TFrame.h"
+#include "TH1.h"
+#include "TError.h"
+#include "TROOT.h"
+
+void writedata(TFile &file)
+{
+   auto h = new TH1F("h", "", 100, 0, 10);
+   h->FillRandom("gaus");
+
+   auto c = new TCanvas("c1");
```

Comment:
Can you rename c1..c3 to something more telling, like "frameThatGetsDeleted" etc? I'm afraid of the complexity of this test: loops, classes, output that's sometimes expected and sometimes not, etc. Not a lot of fun to debug.

Reply:
I renamed the canvas.  Unfortunately, I don't see a much better way to test all the cases.